### PR TITLE
Extract Aspire.Hosting.AWS.Tests

### PR DESCRIPTION
**Pull request by sebastienros**
Part of https://github.com/dotnet/aspire/issues/4294
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5281)

### CODE REVIEW

File: tests/Aspire.Hosting.AWS.Tests/AWSSchemaTests.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+using Xunit;
+
+namespace Aspire.Hosting.AWS.Tests;
+
+public class AWSSchemaTests
+{
+    [Fact]
+    public void ValidateAwsStackSamples()
+    {
+        {
```
Comment (radical) [inline tests/Aspire.Hosting.AWS.Tests/AWSSchemaTests.cs:15]: What is the extra block for?

Comment (sebastienros) [reply tests/Aspire.Hosting.AWS.Tests/AWSSchemaTests.cs:15]: Checking if reviewers do their job

File: tests/Aspire.Hosting.AWS.Tests/AWSSchemaTests.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+using Amazon;
+using Aspire.Hosting.Tests.Schema;
+using Xunit;
+
+namespace Aspire.Hosting.AWS.Tests;
+
+public class AWSSchemaTests
+{
+    [Fact]
+    public void ValidateAwsStackSamples()
```
Comment (radical) [inline tests/Aspire.Hosting.AWS.Tests/AWSSchemaTests.cs:13]: I know other tests share the name, but the name is kinda misleading. `ValidateApplicationSamples` is only validating the generated manifest, and there is no "sample" project as such. Would it be better to name the tests after `AddAWSCloudFormationStack`/`AddAWSClouldFormationTemplate`?

Comment (sebastienros) [reply tests/Aspire.Hosting.AWS.Tests/AWSSchemaTests.cs:13]: Thanks, was wondering what name to use.

**Review by radical — APPROVED** [review]

File: tests/Aspire.Hosting.AWS.Tests/Aspire.Hosting.AWS.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>$(NetCurrent)</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\src\Aspire.Hosting.AWS\Aspire.Hosting.AWS.csproj" IsAspireProjectResource="false" />
+    <ProjectReference Include="..\..\src\Aspire.Hosting.Testing\Aspire.Hosting.Testing.csproj" />
```
Comment (eerhardt) [inline tests/Aspire.Hosting.AWS.Tests/Aspire.Hosting.AWS.Tests.csproj:8]:
```suggestion
```

What's this used for?

File: src/Aspire.Hosting.AWS/Aspire.Hosting.AWS.csproj [hunk 4]
```diff
@@ -26,7 +26,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <InternalsVisibleTo Include="Aspire.Hosting.Tests" />
+    <InternalsVisibleTo Include="Aspire.Hosting.AWS.Tests" />
```
Comment (eerhardt) [inline src/Aspire.Hosting.AWS/Aspire.Hosting.AWS.csproj:29]: What do we use internals for?

**Review by eerhardt — APPROVED** [review]
Just some questions. LGTM

File: src/Aspire.Hosting.AWS/Aspire.Hosting.AWS.csproj (hunk 4 above)
Comment (sebastienros) [reply src/Aspire.Hosting.AWS/Aspire.Hosting.AWS.csproj:29]: image

**Pull request merged**